### PR TITLE
Edit and Delete User Function

### PR DESCRIPTION
**Pull request by karlpandacan**

### CODE REVIEW

File: app/Http/Controllers/Supervisor/UserController.php [hunk 1]
```diff
@@ -107,9 +108,15 @@ public function edit($id)
      * @param  int $id
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, $id)
+    public function update(UserRequest $request, User $user)
     {
-        //
+        try {
+            $userType = $this->userRepository->listUserType();
```
Comment (wataridori) [inline app/Http/Controllers/Supervisor/UserController.php]: Is there something wrong? Where is the update process?

File: app/Http/Controllers/Supervisor/UserController.php [hunk 2]
```diff
@@ -107,9 +108,17 @@ public function edit($id)
      * @param  int $id
      * @return \Illuminate\Http\Response
      */
-    public function update(Request $request, $id)
+    public function update(UserRequest $request, User $user)
     {
-        //
+        try {
+            if($request->hasFile('image'))
```
Comment (wataridori) [inline app/Http/Controllers/Supervisor/UserController.php]: Use `{}` with `if` command

File: app/Http/Requests/UserRequest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                    'email' => 'required|email|max:255|unique:users',
+                    'password' => 'required|confirmed|min:6'
+                ];
+            }
+            case 'PUT':
+            case 'PATCH':
+            {
+                return [
+                    'name' => 'required|max:255',
+                    'email' => 'required|email|max:255|unique:users,email,' . $this->user->id. ',id',
```
Comment (wataridori) [inline app/Http/Requests/UserRequest.php]: Add `space` between `$this->user->id` and `.`

**Comment by wataridori** [comment]
OK

**Comment by nguyenthanhtung88** [comment]
OK

**Pull request merged**